src/main.cpp

```diff
@@ -4384,15 +4418,15 @@ bool TestBlockValidity(CValidationState& state, const CChainParams& chainparams,
     CBlockIndex indexDummy(block);
     indexDummy.pprev = pindexPrev;
     indexDummy.nHeight = pindexPrev->nHeight + 1;
-    // JoinSplit proofs are verified in ConnectBlock
-    auto verifier = ProofVerifier::Disabled();
 
+    auto verifier = ProofVerifier::Disabled();
```

Comment:
I do not understand why this is part of a commit diff.

Reply:
I had previously removed the `CheckBlock` call entirely, since it's redundant with respect to the call in `ConnectBlock` that happens on line 4431, which also calls `CheckBlock`; when I reintroduced it, I did so manually and didn't reintroduce the comment.

An idea that just occurred to me (which would involve a different PR to achieve) is that it might be good if CheckBlock were designed as a wrapper around the CBlock value that could accumulate state and then just inspect that state to avoid redundant checks. It would also give us a way to make it so that "later" checks like ContextualCheckBlock could subsume the "earlier" checks like CheckBlock- right now we can't have the former just call the latter, but if these were instead stateful validations we could do so. We could even take a type-directed approach, such that each stage of checking returned a new type, on which only the "remaining" checks could be invoked.